keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,1699 @@
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Pods cannot be scheduled which makes the feature unavailable for new workloads.
+Existing workloads continue to work without interruption (with respect to this
+feature).
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+TBD.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+TBD.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Pod annotations
+
+Instead of updating CRI and Kubernetes API in lock-step, the API changes could
+be split into two phases. Similar to e.g. how seccomp support was added, Pod
+annotations could be used as an initial user interface before the Kubernetes
+API changes are merged.
+
+Specifically, annotations for specifying RDT and blockio class would be
+supported. These are the two types of QoS-class resources that already have
+basic support in the container runtimes.
+
+- `rdt.resources.alpha.kubernetes.io/default` for setting a Pod-level default RDT
+  class for all containers
+- `rdt.resources.alpha.kubernetes.io/container.<container-name>` for
+  container-specific RDT class settings
+  blockio class for all containers
+- `blockio.resources.alpha.kubernetes.io/default` for setting a Pod-level default
+  blockio class for all containers
+- `blockio.resources.alpha.kubernetes.io/container.<container-name>` for
+  container-specific blockio class settings
```

Comment:
BTW, we don't put “alpha” or ”beta” in annotation keys.

Reply:
K, noted

Reply:
Dropped `alpha`